File: LeetCode/7.ReverseInteger/main.py
from Solution import Solution

def main():
	solution = Solution()
	testCases = [
		#0,
		#123,
		#-123,
		192233720368547758071,
		1000000003,
		#10,
		#1000,
	]
	for test in testCases:
		print solution.reverse(test)
main()

File: LeetCode/7.ReverseInteger/main.c
#include<stdio.h>

int reverse(int x) {
	int result = 0;
	int sign = 1;
	if(x < 0) {
		x = -x;
		sign = -1;
	}
	while(x > 0) {
		if(result * 10 / 10 != result) {
		return 0;
		}
		result = result * 10 + x % 10;
		x /= 10;
	}
	return sign * result;
}

int main() {
	int test = 1534236469;	
	printf("%d", reverse(test));
}
